src/pages/ExamsPage/components/AddAllowanceModal.jsx

```diff
@@ -0,0 +1,239 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ModalDialog,
+  Form,
+  ActionRow,
+  StatefulButton,
+} from '@openedx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import * as constants from '../../../data/constants';
+import { useExamsData, useCreateAllowance, useButtonStateFromRequestStatus } from '../hooks';
+import messages from '../messages';
+
+const AddAllowanceModal = ({ isOpen, close }) => {
+  const { examsList } = useExamsData();
+  const proctoredExams = [];
+  const timedExams = [];
+
+  for (let i = 0; i < examsList.length; i++) { // maybe move to utils
+    const exam = examsList[i];
+    if (exam.examType === 'proctored') {
+      proctoredExams.push(exam);
+    } else {
+      timedExams.push(exam);
+    }
+  }
+
+  const defaultExamType = proctoredExams.length > 0 ? 'proctored' : 'timed';
+  const defaultExamsList = defaultExamType === 'proctored' ? proctoredExams : timedExams;
+
+  const [displayExams, setDisplayExams] = useState(defaultExamsList);
+  const [form, setForm] = useState({
+    'allowance-type': 'additional-minutes',
+    'exam-type': defaultExamType,
+    exams: {},
+  });
+  const [learnerFieldError, setLearnerFieldError] = useState(false);
+  const [examFieldError, setExamFieldError] = useState(false);
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+  const createAllowanceRequestStatus = useButtonStateFromRequestStatus(constants.RequestKeys.createAllowance);
+  const createAllowance = useCreateAllowance();
+  const { formatMessage } = useIntl();
+
+  const addAllowanceButtonProps = {
+    labels: {
+      default: formatMessage(messages.addAllowanceButtonDefaultLabel),
+      pending: formatMessage(messages.addAllowanceButtonPendingLabel),
+      complete: formatMessage(messages.addAllowanceButtonCompleteLabel),
+      error: formatMessage(messages.addAllowanceButtonErrorLabel),
+    },
+    variant: 'primary',
+  };
```

Comment:
Nit: If this is only used in the button below, why splitting the props? I think having this in the component along with its other props it would be easier to read.

Reply:
This was copied from another component that uses this button, I'll refactor so the props aren't split!